meta-integrity/recipes-core/initrdscripts/files/init.ima

```diff
@@ -96,6 +96,21 @@ fi
     print_info "IMA is not enabled. Exiting ..." && exit 2
 
 ima_keyring_id=0x`grep '\skeyring\s*\.ima: ' "${ROOT_DIR}/proc/keys" | awk '{ print $1 }'`
+secondary_keyring_id=0x`grep '\skeyring\s*\.secondary_trusted_keys: ' "${ROOT_DIR}/proc/keys" | awk '{ print $1 }'`
+
+# If we have a secondary trusted keyring, here is the opportunity to load
+# additional trusted keys from the real rootfs.
+if [ "$secondary_keyring_id" != "0x" ]; then
+    for cert in ${ROOT_DIR}/etc/keys/x509_secondary_*.der; do
+        [ ! -s "$cert" ] && continue
+
+        if ! evmctl import "$cert" "$secondary_keyring_id" >"${ROOT_DIR}/dev/null"; then
```

Comment:
Using evmctl is working. But it is better to use `keyctl padd asymmetric "Secondary system trusted key" %:.secondary_trusted_keys < $cert` here. This can remove the complexity of parsing keyring id.

Reply:
OK, I can re-work things to do a general clean-up first.